src/workflows/airqo_etl_utils/weather_data_utils.py

```diff
@@ -189,21 +190,21 @@ def aggregate_data(data: pd.DataFrame) -> pd.DataFrame:
             station_group = station_group.sort_index(axis=0)
 
             averaging_data = station_group.copy()
-            del averaging_data["precipitation"]
-            averages = pd.DataFrame(averaging_data.resample("H").mean())
-            averages["timestamp"] = averages.index
+            averaging_data.drop(columns=["precipitation"], inplace=True)
+            numeric_cols = averaging_data.select_dtypes(include=[np.number]).columns
+            averages = averaging_data.resample("H")[numeric_cols].mean()
             averages.reset_index(drop=True, inplace=True)
 
-            summing_data = station_group.copy()[["timestamp", "precipitation"]]
+            summing_data = station_group.copy()[["precipitation"]]
             sums = pd.DataFrame(summing_data.resample("H").sum())
             sums["timestamp"] = sums.index
             sums.reset_index(drop=True, inplace=True)
 
-            merged_data = pd.merge(left=averages, right=sums, on="timestamp")
+            merged_data = pd.concat([averages, sums], axis=1)
```

Comment:
Hi @MartinKalema 

I believe it would be best for us to stick with `pd.merge` as it it combines data frames based on specified columns. `pd.concate` deals with indices. what do you think about this?

Reply:
Also @AnthonyByansi `pd.merge` is deprecated and will be removed from consecutive releases. @MartinKalema review the data to ensure accuracy vs what is required/needed.

Reply:
Thanks for the clarification @NicholasTurner23

Reply:
Hi @MartinKalema , awaiting your response to the status of this thread. Is every feedback item addressed? Are we good to go now?

Reply:
Hi @Baalmart, Yes, everything was addressed and we are good to go.